lib/dialogs/buy/buy_dialog.dart

```diff
@@ -539,26 +541,31 @@ class _BuyDialogState extends State<BuyDialog> {
                     ],
                   ),
                 ),
+
                 SizedBox(
                   width: wid,
                   child: Row(
                     mainAxisAlignment: MainAxisAlignment.center,
                     children: [
-                      AthleteBuyApproveButton(
-                        width: 175,
-                        height: 40,
-                        text: 'Approve',
-                        amountInputted: _aptAmountController.text,
-                        aptBuyInfo: aptBuyInfo,
-                        athlete: widget.athlete,
-                        aptName: widget.athleteName,
-                        aptId: widget.athleteId,
-                        longOrShort: aptLongOrShort,
-                        approveCallback: bloc.swapController.approve,
-                        confirmCallback: bloc.swapController.swap,
-                        confirmDialog: transactionConfirmed,
-                        walletAddress: userWalletAddress.walletAddress,
-                      ),
+                      if (bloc.state.status != BlocStatus.error) ...[
+                        AthleteBuyApproveButton(
+                          width: 175,
+                          height: 40,
+                          text: 'Approve',
+                          amountInputted: _aptAmountController.text,
+                          aptBuyInfo: aptBuyInfo,
+                          athlete: widget.athlete,
+                          aptName: widget.athleteName,
+                          aptId: widget.athleteId,
+                          longOrShort: aptLongOrShort,
+                          approveCallback: bloc.swapController.approve,
+                          confirmCallback: bloc.swapController.swap,
+                          confirmDialog: transactionConfirmed,
+                          walletAddress: userWalletAddress.walletAddress,
+                        ),
+                      ] else ...[
+                        const WarningTextButton(warningTitle: 'Error on Action')
+                      ]
```

Comment:
I think this is a poor approach because it replaces the original action which gives the user no chance to retry.
Showing error should follow a cleaner pattern,  in our case using a toast, e.g.: `context.showWalletWarningToast()`.

Also, if you need to return a single widget you don't need to spread a new list with 1 element, you can just specify the widget directly. You need the spread list for more than 1 elements.

Reply:
I implemented to show error toast using BlocListener